superset/common/query_context_processor.py

```diff
@@ -314,30 +316,49 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
         rv_dfs: List[pd.DataFrame] = [df]
 
         time_offsets = query_object.time_offsets
-        outer_from_dttm = query_object.from_dttm
-        outer_to_dttm = query_object.to_dttm
+        outer_from_dttm, outer_to_dttm = get_since_until_from_query_object(query_object)
+        if not outer_from_dttm or not outer_to_dttm:
+            raise QueryObjectValidationError(
+                _(
+                    "An enclosed time range (both start and end) must be specified "
+                    "when using a Time Comparison."
+                )
+            )
```

Comment:
Out of scope for this PR, but I believe we could remove this requirement, as I feel we should be able to do time comparison without specifying a time range (I usually work around this by placing a `100 years ago : now` on the time filter)